ffmpeg_tools/validation.py

```diff
@@ -402,20 +431,30 @@ def validate_audio_sample_rates(
 
 def validate_video_codec_conversion(src_codec, dst_codec):
     codec = codecs.VideoCodec(src_codec)
+
     if dst_codec not in codec.get_supported_conversions():
         raise UnsupportedVideoCodecConversion(src_codec, dst_codec)
+
+    if codecs.VideoCodec(dst_codec).get_encoder() is None:
+        raise MissingVideoEncoder(video_codec=dst_codec)
+
     return True
 
 
-def validate_audio_codec_conversion(src_codec, dst_codec, audio_stream):
+def validate_audio_codec_conversion(src_codec, dst_codec, src_channel_count):
     codec = codecs.AudioCodec(src_codec)
+
     if dst_codec not in codec.get_supported_conversions():
         raise UnsupportedAudioCodecConversion(src_codec, dst_codec)
-    if src_codec != dst_codec and \
-            audio_stream['channels'] > _MAX_SUPPORTED_AUDIO_CHANNELS:
+
+    if codecs.AudioCodec(dst_codec).get_encoder() is None:
+        raise MissingAudioEncoder(audio_codec=dst_codec)
```

Comment:
Validation shouldn't fail in case of target codec. It should assume, that parameter doesn't change, if not specified. Related to review of https://github.com/golemfactory/golem/pull/4954#pullrequestreview-326881496

Reply:
It won't fail. `validate_transcoding_params()` won't run this function if the target codec is not known and can't be inferred from muxer info:
``` python
        dest_audio_codec = _get_dst_audio_codec(dst_params, dst_muxer_info)
        if dest_audio_codec is not None:
            validate_audio_codecs(dst_params["container"], [dest_audio_codec])
            for audio_stream in meta.get_streams(src_metadata, 'audio'):
                validate_audio_codec_conversion(
                    audio_stream.get('codec_name'),
                    dest_audio_codec,
                    audio_stream.get('channels'),
                )
```